xarray/core/combine.py

```diff
@@ -605,21 +560,141 @@ def auto_combine(datasets, concat_dim=_CONCAT_DIM_DEFAULT,
     --------
     concat
     Dataset.merge
-    """  # noqa
-
-    # Coerce 1D input into ND to maintain backwards-compatible API until API
-    # for N-D combine decided
-    # (see https://github.com/pydata/xarray/pull/2553/#issuecomment-445892746)
-    if concat_dim is None or concat_dim == _CONCAT_DIM_DEFAULT:
-        concat_dims = concat_dim
-    elif not isinstance(concat_dim, list):
-        concat_dims = [concat_dim]
+    """
+
+    if concat_dim is '_not_supplied':
+        concat_dim = _CONCAT_DIM_DEFAULT
+    else:
+        message = """In xarray version 0.14 `auto_combine` will be deprecated,
+                  and `open_mfdataset` will no longer accept a `concat_dim`
+                  argument. To get equivalent behaviour from now on please use
+                  the new `combine_manual` function instead (or the
+                  `combine='manual'` option to open_mfdataset)."""
+        warnings.warn(message, FutureWarning)
+
+    if _dimension_coords_exist(datasets):
+        message = """In xarray version 0.14 `auto_combine` will be deprecated.
+                  The datasets supplied have global dimension coordinates.
+                  You may want to use the new `combine_auto` function (or the
+                  `combine='auto'` option to `open_mfdataset` to order the
+                  datasets before concatenation. Alternatively, to continue
+                  concatenating based on the order the datasets are supplied in
+                  in future, please use the new `combine_manual` function (or
+                  the `combine='manual'` option to open_mfdataset)."""
+        warnings.warn(message, FutureWarning)
     else:
-        concat_dims = concat_dim
-    infer_order_from_coords = False
-
-    # The IDs argument tells _auto_combine that the datasets are not yet sorted
-    return _auto_combine(datasets, concat_dims=concat_dims, compat=compat,
-                         data_vars=data_vars, coords=coords,
-                         infer_order_from_coords=infer_order_from_coords,
-                         ids=False)
+        message = """In xarray version 0.14 `auto_combine` will be deprecated.
+                  The datasets supplied do not have global dimension
+                  coordinates. In future, to continue concatenating without
+                  supplying dimension coordinates, please use the new
+                  `combine_manual` function (or the `combine='manual'` option
+                  to open_mfdataset)."""
+        warnings.warn(message, FutureWarning)
+
+    if _requires_concat_and_merge(datasets):
```

Comment:
It looks like up to three different FutureWarnings could be issued when a user calls this function.

I think it's fantastic how you attempt to diagnose the issue to give the user the best possible guidance, but could we consolidate all this into a single warning, e.g., by building up the warning message dynamically?

Reply:
> could we consolidate all this into a single warning?

Done. It is potentially a pretty long warning, so I'm not sure if that's much better though.